src/updaters/ocaml/esy-json.ts

```diff
@@ -0,0 +1,45 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {checkpoint, CheckpointType} from '../../util/checkpoint';
+import {Update, UpdateOptions, VersionsMap} from '../update';
+import {GitHubFileContents} from '../../github';
+
+export class EsyJson implements Update {
+  path: string;
+  changelogEntry: string;
+  version: string;
+  versions?: VersionsMap;
+  packageName: string;
+  create: boolean;
+  contents?: GitHubFileContents;
+
+  constructor(options: UpdateOptions) {
+    this.create = false;
+    this.path = options.path;
+    this.changelogEntry = options.changelogEntry;
+    this.version = options.version;
+    this.packageName = options.packageName;
+  }
+
+  updateContent(content: string): string {
```

Comment:
If I'm understanding correctly, we're going to attempt to update every `.json` file in the repository, even though some might not be `esy` files?

Could we add a helper that attempts to detect whether it's the appropriate file type, by looking at some well known fields.

Reply:
Yeah, the weirdness with esy is that you can have different json files for different sanboxes, those might contain an `esy` section, `overrides` section, or none of them (and just look like a package.json file...). Maybe we just check for the `version` key and update it? Even if an esy package contains an `esy.json` and `package.json`, it will be common that they both need to be updated anyway.

Thoughts?

Reply:
If you think the version will be okay, I think that's a good start (we'll at least get things working for your needs).

My only concern is as other folks with fixtures, etc., start to use your release strategy -- maybe at that point we can introduce an option to pass a list of files rather than using search?